core/src/client/async_client/mod.rs

```diff
@@ -232,11 +231,15 @@ impl Client {
 		*err_lock = next_state;
 		err
 	}
-}
 
-impl Drop for Client {
-	fn drop(&mut self) {
-		self.to_back.close_channel();
+	/// Completes when the client is disconnected or the client's background task encountered an error.
+	/// If the client is already disconnected, the future produced by this method will complete immediately.
+	///
+	/// # Cancel safety
+	///
+	/// This method is cancel safe.
+	pub async fn notify_on_disconnect(&self) {
```

Comment:
I think I suggested this name but I find it a bit long.

Maybe `on_disconnect` or `disconnect` are better.